### ISSUE TITLE
Disable public user registration

### ISSUE THREAD

**Issue by tmelikoff** [wallabag/wallabag#1873]
It would be really useful to disable public user registration and allowing only the administrator to create new accounts. I believe that some of us use wallabag only for personal use (or family).

Also, if such option (configuration file or switch in control panel) were enabled, registration button should be hidden in the homepage.

Thank you.

**Comment by NerosTie** [comment]
Until there is a solution and if you use Nginx, add this in your configuration:

```
    location ^~ /register {
      deny all;
      return 403;
    }
```

**Comment by ebal** [comment]
This is the work-around for httpd-2.2.15

     <LocationMatch "/register">
         Deny from All
     </LocationMatch>

**Comment by fabsh** [comment]
Uhm. This is a rather important feature, I would think. I'm guessing a lot of people install this app to for their own use only. I would like to be able to prevent other people from using my server resources. Version 2.x suggest production ready to me. But an app with such a glaring oversight can hadly be used in production...

**Comment by nicosomb** [comment]
@j0k3r what is the best way? A parameter (true/false)? If true, we hide the register button and we override the register route like here http://stackoverflow.com/a/32332378?

**Comment by j0k3r** [comment]
@nicosomb Yeah could a good idea. An extra parameter in internal configuration.

**Comment by nicosomb** [comment]
I think it's the easier way. OK, I change the milestone.

**Comment by nicosomb** [comment]
Done by @tcitworld in #2186 👍 .

**Comment by danicotillas** [comment]
@nicosomb I can´t find this option on the Configuration panel, How can i disable it?

**Comment by tcitworld** [comment]
@danicotillas It's in the `app/config/parameters.yml` file, named `fosuser_registration`.

**Comment by danicotillas** [comment]
@tcitworld Thanks, I change it to "false" but it still appears... What are the changes I should make?

**Comment by tcitworld** [comment]
Clear the cache `bin/console cache:clear -e prod`

**Comment by ecodiv** [comment]
I have changed the parameters.yml as suggested. But how do I clear the cache using the above? Do I need command line access I have Wallabag installed on a shared webhosting site, installed via Softaculous script installer)?

**Comment by ecodiv** [comment]
OK, got it, doing:

1) Open the file in the folder `app/config/parameters.yml`
2) Set `fosuser_registration = false`
3) connect to your server using ssh (command line)
4) Go to the root folder of your Wallabag installation
5) Run `php bin/console cache:clear -e prod`

Would still be nice to have this as a click-able option on the internal configuration panel.

**Comment by viktormadarasz** [comment]
Hi @ecodiv 
I tried the same on my docker instance but it keeps regenerating the parameters.yml with the value true

**Comment by STaRDoGG** [comment]
How come this isn't added as a simple check-box in the options?

Also, why isn't there a simple button/link in options to clear the cache? lol

**Comment by burkemw3** [comment]
The [Aug 29, 2017](https://github.com/wallabag/wallabag/issues/1873#issuecomment-325670552) comment sent me on a wild-goose chase on getting this working in docker. For docker, you only actually need to pass the SYMFONY__ENV__FOSUSER_REGISTRATION environment variable, which is documented in the docker readme.

**Comment by j0k3r** [comment]
@burkemw3 Do we need a better documentation regarding this point?

**Comment by burkemw3** [comment]
@j0k3r I don't think so. This issue is my top google search result for "wallabag disable register", and the issue didn't have the full story. So, I think my comment provides enough to close the loop on this without other documentation.

**Comment by jalcine** [comment]
Nah, this is something that _should_ be documented. Granted, this came up in DuckDuckGo but relying on search engines to document something that _should_ have a user toggle (same way the user registration via API is a thing) isn't the way to user friendliness.

**Comment by j0k3r** [comment]
@jalcine How can this be more documented?
That option is defined in the Docker readme: https://github.com/wallabag/docker

**Comment by stevelord** [comment]
The problem is there are config entries disclosed around different parts of the Internet but no links to how they would be used. I found this github issue and @j0k3r's comment via an internet search for how to disable public registrations, which took me to the docker repo. In the docs, I have to search for "enable public registrations" to find out how to disable public registrations.

Likewise I've been trying to find out what the default username and password are for the admin account, when my mistake was not searching through every instance of the word "wallabag" because the default admin user and password is "wallabag". I still don't know where in the documentation to find this. I actually figured this out on my migrated install today by using mysql and guessing.

Wallabag's docs could really do with a task-oriented index and an update to the installation guide featuring common post-install steps.
